curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -0,0 +1,149 @@
+---
+id: 66db529d37ad966480ebb633
+title: Build a Lunch Picker Program
+challengeType: 14
+dashedName: build-a-lunch-picker-program
+---
+
+# --description--
+
+In this lab, you will create a program that will help you decide what to eat for lunch. You will have two arrays with different food items. You will use the `push`, `pop` and `shift` methods to modify the arrays. You will also use the `Math.random()` method to generate a random number that will be used to select a random meal from the arrays.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should create a `kidsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+2. You should create an `adultsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+3. Once items have been added to the `kidsMeal`, you should remove the last item from it using `pop()`.
+4. Once items have been added to the `adultsMeal`, you should remove the first item from it using `shift()`.
+5.  Generate a random number between `0` to the length of the `kidsMeal` array using `Math.random()` and store it in a variable called `randomIndexKids`.
+6. Generate a random number between `0` to the length of the `adultsMeal` array using `Math.random()` and store it in a variable called `randomIndexAdults`.
+7.  Log the following messages to the console:
+
+- `"Today kids should eat:"` followed by the selected meal to the console.
+- `"Today adults should eat:"` followed by the selected meal to the console.
+
+# --hints--
+
+You should declare a `kidsMeal` array.
+
+```js
+assert.isArray(kidsMeal);
+```
+
+You should declare an `adultsMeal` array.
+
+```js
+assert.isArray(adultsMeal);
+```
+
+Your `kidsMeal` array should contain atleast three items.
+
+```js
+assert.isAtLeast(kidsMeal.length, 2);
```

Comment:
```suggestion
assert.isAtLeast(kidsMeal.length, 3);
```

Reply:
I have kept this intentionally as this would be the length after using `pop()` and `shift()` on the respective arrays with length `3`.

Reply:
This should check that the array contains at least 3 elements at the moment of the assignment. Since it is modified, I'm afraid we should use regex for this test. Something similar to the following?

```suggestion
const matched = code.match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
const kidsMealArray = matched.groups.items.split(",");
assert.isAtLeast(kidsMealArray.length, 3);
```

Reply:
to avoid any issues with comments I think they should be removed any time a regex test is used

Reply:
Suggestion updated following your comment @ilenia-magoni

Reply:
Thank you @Dario-DC for the suggestion, this works perfectly.
I have updated other tests too.